newrelic/api/log.py

```diff
@@ -69,11 +55,46 @@ def safe_json_encode(obj, ignore_string_types=False, **kwargs):
 class NewRelicContextFormatter(logging.Formatter):
     DEFAULT_LOG_RECORD_KEYS = frozenset(set(vars(logging.LogRecord("", 0, "", 0, "", (), None))) | {"message"})
 
-    def __init__(self, *args, **kwargs):
-        super(NewRelicContextFormatter, self).__init__()
+    def __init__(self, *args, stack_trace_limit: Optional[int] = 0, **kwargs):
+        """
+        :param Optional[int] stack_trace_limit:
+            Specifies the maximum number of frames to include for stack traces.
+            Defaults to `0` to suppress stack traces.
+            Setting this to `None` will make it so all available frames are included.
+        """
+        super(NewRelicContextFormatter, self).__init__(*args, **kwargs)
+
+        if stack_trace_limit is not None:
+            if not isinstance(stack_trace_limit, int):
+                raise TypeError("stack_trace_limit must be None or a non-negative integer")
+            elif stack_trace_limit < 0:
+                raise ValueError("stack_trace_limit must be None or a non-negative integer")
+        self._stack_trace_limit = stack_trace_limit
+
+    def format_exc_info(self, exc_info):
+        _, _, fullnames, message = parse_exc_info(exc_info)
+        fullname = fullnames[0]
+
+        formatted = {
+            "error.class": fullname,
+            "error.message": message,
+        }
 
-    @classmethod
-    def log_record_to_dict(cls, record):
+        expected = is_expected_error(exc_info)
```

Comment:
An issue here, we're taking `log_record_to_dict()` which is a classmethod and changing it to an instance method. This would be a breaking change to the public API.

The same goes for `format_exc_info()` changing from a top level function to an instance method.

We definitely have customers who use these functions, or at least `log_record_to_dict()` without instantiating a class for convenience.

I'll push a commit with my proposal, but I'd like to convert these both into class functions, and export the names to the top level of the module so they can be used like functions. I think passing in the stack trace limit as an argument to the functions should preserve the most backwards compatibility for other users.